vertx-web/src/main/java/io/vertx/ext/web/sstore/AbstractSession.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.ext.web.sstore;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.PRNG;
+import io.vertx.ext.web.Session;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author <a href="mailto:plopes@redhat.com">Paulo Lopes</a>
+ */
+public abstract class AbstractSession implements Session {
+
+  private static final char[] HEX = "0123456789abcdef".toCharArray();
+
+  private PRNG prng;
+
+  private String id;
+  private long timeout;
+  private volatile Map<String, Object> data;
+  private long lastAccessed;
+  private int version;
+
+  protected void setId(String id) {
+    this.id = id;
+  }
+
+  protected void setTimeout(long timeout) {
+    this.timeout = timeout;
+  }
+
+  protected void setData(Map<String, Object> data) {
+    if (data != null) {
+      this.data = data;
+      this.crc = checksum();
+    }
+  }
+
+  protected void setData(JsonObject data) {
+    if (data != null) {
+      setData(data.getMap());
+    }
+  }
+
+  protected void setLastAccessed(long lastAccessed) {
+    this.lastAccessed = lastAccessed;
+  }
+
+  protected void setVersion(int version) {
+    this.version = version;
+  }
+
+  // state management
+  private boolean destroyed;
+  private boolean renewed;
+  private String oldId;
+  private int crc;
+
+  /**
+   * Important note: This constructor (even though not referenced anywhere) is required for serialization purposes. Do
```

Comment:
You could simply say this is mandatory as per `ClusterSerializable` contract

Reply:
done